core/controllers/voice_artist_test.py

```diff
@@ -416,3 +418,188 @@ def test_cannot_deassign_random_user_from_voice_artist(self):
+            csrf_token=self.csrf_token,
+            expected_status_int=400,
+            upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertEqual(
+            response_dict['error'],
+            'No filename extension: it should have '
+            'one of the following extensions: '
+            '%s' % list(feconf.ACCEPTED_AUDIO_EXTENSIONS.keys()))
+
+    def test_invalid_extension(self):
+        filename_without_extension = 'test'
+        invalid_extension = 'wav'
+        supplied_filename = (
+                '%s.%s' % (filename_without_extension, invalid_extension))
+
+        with utils.open_file(
+                os.path.join(feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_MP3),
+                'rb', encoding=None
+        ) as f:
+            raw_audio = f.read()
+        response_dict = self.post_json(
+            '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+            {'filename': supplied_filename},
+            csrf_token=self.csrf_token,
+            expected_status_int=400,
+            upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertEqual(
+            response_dict['error'],
+            'Invalid filename extension: it should have '
+            'one of the following extensions: %s'
+            % list(feconf.ACCEPTED_AUDIO_EXTENSIONS.keys()))
+
+    def test_mutagen_bad_audio(self):
+        response_dict = self.post_json(
+            '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+            {'filename': 'test.mp3'},
+            csrf_token=self.csrf_token,
+            expected_status_int=400,
+            upload_files=(
+                ('raw_audio_file', 'unused_filename', 'non_audio_data'),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertEqual(
+            response_dict['error'], 'Audio not recognized as a mp3 file')
+
+    def test_detect_non_audio(self):
+        with utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            'rb', encoding=None
+        ) as f:
+            raw_audio = f.read()
+
+        with self.accepted_audio_extensions_swap:
+            response_dict = self.post_json(
+                '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+                {'filename': self.TEST_AUDIO_FILE_FLAC},
+                csrf_token=self.csrf_token,
+                expected_status_int=400,
+                upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
+            )
+        self.assertEqual(
+            response_dict['error'], 'Audio not recognized as a flac file')
+
+    def test_too_long(self):
+        with utils.open_file(
+            os.path.join(
+                feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_OVER_MAX_LENGTH),
```

Comment:
Looks like `TEST_AUDIO_FILE_OVER_MAX_LENGTH` is only used in this test. Should it be a local variable instead?